docs/02_guides/05_scrapy.mdx

```diff
@@ -7,90 +7,99 @@ import CodeBlock from '@theme/CodeBlock';
+The Apify SDK provides an Apify-Scrapy integration. The main challenge of this is to combine two asynchronous frameworks that use different event loop implementations. Scrapy uses [Twisted](https://twisted.org/) for asynchronous execution, while the Apify SDK is based on [asyncio](https://docs.python.org/3/library/asyncio.html). The key thing is to install the Twisted's `asyncioreactor` to run Twisted's asyncio compatible event loop. This allows both Twisted and asyncio to run on a single event loop, enabling a Scrapy spider to run as an Apify Actor with minimal modifications.
 
-:::
+<CodeBlock className="language-python" title="__main.py__: The Actor entry point ">
+    {UnderscoreMainExample}
+</CodeBlock>
 
-Some of the key features of Scrapy for web scraping include:
+In this setup, `apify.scrapy.initialize_logging` configures an Apify log formatter and reconfigures loggers to ensure consistent logging across Scrapy, the Apify SDK, and other libraries. The `apify.scrapy.run_scrapy_actor` bridges asyncio coroutines with Twisted's reactor, enabling the Actor's main coroutine, which contains the Scrapy spider, to be executed.
 
-- **Request and response handling** - Scrapy provides an easy-to-use interface for making HTTP requests and handling responses,
-allowing you to navigate through web pages and extract data.
-- **Robust Spider framework** - Scrapy has a spider framework that allows you to define how to scrape data from websites,
-including how to follow links, how to handle pagination, and how to parse the data.
-- **Built-in data extraction** - Scrapy includes built-in support for data extraction using XPath and CSS selectors,
-allowing you to easily extract data from HTML and XML documents.
-- **Integration with other tool** - Scrapy can be integrated with other Python tools like BeautifulSoup and Selenium for more advanced scraping tasks.
+<CodeBlock className="language-python" title="main.py: The Actor main coroutine">
+    {MainExample}
+</CodeBlock>
 
-## Using Scrapy template
+Within the Actor's main coroutine, the Actor's input is processed as usual. The function `apify.scrapy.apply_apify_settings` is then used to configure Scrapy settings with Apify-specific components before the spider is executed. The key components and other helper functions are described in the next section.
 
-The fastest way to start using Scrapy in Apify Actors is by leveraging the [Scrapy Actor template](https://apify.com/templates/categories/python). This template provides a pre-configured structure and setup necessary to integrate Scrapy into your Actors seamlessly. It includes: setting up the Scrapy settings, `asyncio` reactor, Actor logger, and item pipeline as necessary to make Scrapy spiders run in Actors and save their outputs in Apify datasets.
+## Key integration components
```

Comment:
It would be awesome if we could have a nice diagram here :slightly_smiling_face: In the ideal case, it would show how the asyncio event loop and twisted reactor interact. We can definitely postpone that though.

Reply:
The point is that Asyncio and Twisted share the same event loop. I'm not familiar with the details of how Asyncio and Twisted exactly interact together. Other than that, we have a separate thread with an Asyncio event loop for the Scheduler's synchronous calls.

Reply:
We gave it two weeks and made it work, but that doesn't mean we understand the sorcery enough to draw diagrams 😅 But I think it's something like...


![diagram](https://github.com/user-attachments/assets/e716cab7-fa27-43a8-8400-47033ee81869)

(It's black on transparent SVG, so you probably won't see it well in dark mode)

Reply:
I don't know if I chose the best diagram for this, maybe https://swimlanes.io/ would be better